### PR TITLE
LPD-35535 - First implementation of getting an existing widget page via Headless API

### PR DESCRIPTION
**Pull request by ruben-pulido**
https://liferay.atlassian.net/browse/LPD-35535 - First implementation of getting an existing widget page via Headless API

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

File: test.properties [hunk 1]
```diff
@@ -6880,7 +6880,8 @@ test.batch.run.property.query[functional-orcllinux9-tomcat90-mysql57-jdk8]=\
         **/asset/asset-publisher-test/**/*Test.java,\
         **/asset/asset-publisher-web/**/*Test.java,\
         **/fragment/**/*Test.java,\
-        **/headless/headless-delivery/**/*SitePage*Test.java,\
+        **/headless/headless-admin-site/**/*Test.java,\
```
Comment (ealonso) [inline test.properties:6883]: @ruben-pulido Why you are adding back ContentSetElement to page management routine?

Comment (ealonso) [reply test.properties:6883]: Also, Why are you removing SitePage*Test.java?

File: modules/apps/headless/headless-admin-site/headless-admin-site-impl/src/main/java/com/liferay/headless/admin/site/internal/resource/v1_0/SitePageResourceImpl.java [hunk 2]
```diff
@@ -18,4 +28,54 @@
...
+		throws Exception {
+
+		Group group = _groupLocalService.getGroupByExternalReferenceCode(
+			siteExternalReferenceCode, contextCompany.getCompanyId());
+
+		return _sitePageDTOConverter.toDTO(
+			new DefaultDTOConverterContext(
+				contextAcceptLanguage.isAcceptAllLanguages(), null,
+				_dtoConverterRegistry, contextHttpServletRequest,
+				_layoutService.getLayoutByExternalReferenceCode(
```
Comment (ealonso) [inline modules/apps/headless/headless-admin-site/headless-admin-site-impl/src/main/java/com/liferay/headless/admin/site/internal/resource/v1_0/SitePageResourceImpl.java:58]: @ruben-pulido I think we should extract it into a new variable

File: test.properties (hunk 1 above)
Comment (ruben-pulido) [reply test.properties:6883]: This was a mistake. Thanks for catching it.

File: modules/apps/headless/headless-admin-site/headless-admin-site-impl/src/main/java/com/liferay/headless/admin/site/internal/resource/v1_0/SitePageResourceImpl.java (hunk 2 above)
Comment (ruben-pulido) [reply modules/apps/headless/headless-admin-site/headless-admin-site-impl/src/main/java/com/liferay/headless/admin/site/internal/resource/v1_0/SitePageResourceImpl.java:58]: Thanks!

**Comment by ruben-pulido** [comment]
https://github.com/liferay-echo/liferay-portal/pull/16243

**Pull request closed without merging**